website/docs/concepts/immutability/why_immutability_example.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'why_immutability_example.g.dart';
+
+class Fruit {
+  Fruit({this.typeOfFruit = 'Apple'});
+
+  String typeOfFruit;
+}
+
+@riverpod
+Fruit getFruit(GetFruitRef ref) {
+  return Fruit();
+}
+
+class ChangeFruitButton extends ConsumerWidget {
+  const ChangeFruitButton({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    ref.listen<Fruit>(
+      getFruitProvider,
+      (Fruit? previousFruit, Fruit newFruit) {
+        print('The type of fruit has changed');
+      },
+    );
+    return ElevatedButton(
+      onPressed: () {
+        var fruit = ref.read<Fruit>(getFruitProvider);
+        fruit.typeOfFruit = 'Banana'; // Does not trigger listener
```

Comment:
I think it would make more sense to showcase the issue with ref.select and ref.listen's "previous" when using a mutable state.

Not triggering listeners isn't really a mutability issue, as folks could be using some form of `notifyListeners`.

Reply:
I went to illustrate the issue with ref.listen's "previous", but even after reading the docs, I'm not sure what the issue is. Could you explain the issue here so I can include it in the code example?

As for `ref.select`, let me know if Line 52 of [the code example](https://github.com/rrousselGit/riverpod/blob/7278238dfe9faa3f31c1d6befea4e0d033e29830/website/docs/immutability/raw_why_immutability_example.dart) captures this scenario well enough:

Reply:
It looks like we're [already referencing the issue](https://github.com/rrousselGit/riverpod/blob/e9840394f515d6b9fc2dbed351d67f3cc39e4e28/website/docs/immutability.mdx) you're mentioning, in this PR:

> Using immutable state on a `Notifier` is also necessary if you want `ref.listen`'s "previous" to work. If you use mutable state instead, **the "previous" and "next" values on change could end-up being the same if an object is mutated**.

I'm just not sure in what scenario the previous and next values could be the same. Once I know, I can include that in the code example.